test/es-module/test-esm-double-encoding.mjs

```diff
@@ -0,0 +1,3 @@
+// Flags: --experimental-modules
+import '../common';
+import './test-esm-double-encoding-native%2520.js';
```

Comment:
```js
// Flags: --experimental-modules
import '../common';

// Assert we can import files with `%` in their pathname.

import './test-esm-double-encoding-native%20.js';
```
P.S. shouldn't it be `%20` (pre encoded)?

Reply:
> P.S. shouldn't it be `%20` (pre encoded)?

`import` actually uses URLs for module resolution, so `%2520` is read as `%20` (ie this is correct).

Reply:
`%20` gets decoded to ` ` by the resolver, so that's why it is a double-decoding. Will see if I can illustrate this better.